golang/lib/execution/lambda_executor.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2020 - present Kurtosis Technologies LLC.
+ * All Rights Reserved.
+ */
+
+package execution
+
+import (
+	"github.com/kurtosis-tech/kurtosis-client/golang/kurtosis_core_rpc_api_bindings"
+	"github.com/kurtosis-tech/kurtosis-client/golang/lib/networks"
+	"github.com/kurtosis-tech/kurtosis-client/golang/lib/services"
+	"github.com/kurtosis-tech/kurtosis-lambda-api-lib/golang/kurtosis_lambda_docker_api"
+	"github.com/kurtosis-tech/kurtosis-lambda-api-lib/golang/kurtosis_lambda_rpc_api_bindings"
+	"github.com/kurtosis-tech/kurtosis-lambda-api-lib/golang/kurtosis_lambda_rpc_api_consts"
+	"github.com/kurtosis-tech/minimal-grpc-server/server"
+	"github.com/palantir/stacktrace"
+	"google.golang.org/grpc"
+	"time"
+)
+
+const (
+	grpcServerStopGracePeriod = 5 * time.Second
+)
+
+type LambdaExecutor struct {
+	apiContainerSocket        string
+	serializedCustomParamsStr string
+	configurator              LambdaConfigurator
+}
+
+func NewLambdaExecutor(apiContainerSocket string, serializedCustomParamsStr string, configurator LambdaConfigurator) *LambdaExecutor {
+	return &LambdaExecutor{apiContainerSocket: apiContainerSocket, serializedCustomParamsStr: serializedCustomParamsStr, configurator: configurator}
+}
+
+func (executor LambdaExecutor) Run() error {
+
+	lambda, err := executor.configurator.ParseParamsAndCreateLambda(executor.serializedCustomParamsStr)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred parsing the serialized custom params and creating the lambda")
+	}
+
+	var networkCtx *networks.NetworkContext
+	if executor.apiContainerSocket != "" {
```

Comment:
If we flip this condition, the code becomes a lot easier:
```
if executor.apiContainerSocket == "" {
    ....error....
}
```

Reply:
Made this into general style guide: https://github.com/kurtosis-tech/kurtosis-internal-docs/blob/master/general-style-guide.md#default-to-eject-early-eject-often

Reply:
Very useful advice